src/test/java/test/testng1287/CorrectOrderTest.java

```diff
@@ -0,0 +1,38 @@
+package test.testng1287;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+
+import org.junit.Assert;
+import org.testng.ITestNGListener;
+import org.testng.TestNG;
+import org.testng.annotations.Test;
+import org.testng.xml.XmlClass;
+import org.testng.xml.XmlSuite;
+import org.testng.xml.XmlTest;
+import org.testng.xml.XmlSuite.ParallelMode;
+
+import test.InvokedMethodNameListener;
+import test.SimpleBaseTest;
+
+public class CorrectOrderTest extends SimpleBaseTest {
+
+	@Test
+	public void correctOrder() {
+		TestNG tng = create();
+		XmlSuite s = createXmlSuite("CorrectOrder");
+		s.setParallel(ParallelMode.METHODS);
+		XmlTest t = new XmlTest(s);
+
+		t.getXmlClasses().add(new XmlClass("test.testng1287.GroupClass"));

```

Comment:
Prefer `test.testng1287.GroupClass.class` or `test.testng1287.GroupClass.class.getName()` instead.
Otherwise, you have a `createXmlTest` too.